src/base/bittorrent/trackerentry.h

```diff
@@ -58,7 +61,8 @@ namespace BitTorrent
 
         QString url() const;
         QStringList messages() const;
-        Status status() const;
+        TrackerStatus status() const;
+        QList<EndpointView> endpoints() const;
```

Comment:
use `QVector` instead.
also don't forget to call QVector<>::reserve().

Reply:
Older version of Qt doesn't have in place initialization in QVector. And by design this class is not default constructable, so need to have move semantic to work. Works fine with QVector on newer Qt version, but to keep compatibility QList is the way to go.

Reply:
Sorry, I can't find in your answer any strong reason that prevents from using QVector here (it's mostly unrelated).

Reply:
> strong reason

For one it doesn't compile. I made this class default constructible and compatible with old QVector, I'm tired.

Reply:
>For one it doesn't compile.

Well, seems I didn't understand your previous answer... For me, when you say "this" you refer to previously mentioned thing (QVector) but seems you don't.